ray-operator/apis/ray/v1alpha1/rayjob_types.go

```diff
@@ -0,0 +1,90 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type JobStatus string
+
+const (
+	PENDING   JobStatus = "PENDING"
+	RUNNING   JobStatus = "RUNNING"
+	STOPPED   JobStatus = "STOPPED"
+	SUCCEEDED JobStatus = "SUCCEEDED"
+	FAILED    JobStatus = "FAILED"
+)
+
+// RayJobSpec defines the desired state of RayJob
+type RayJobSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	RuntimeEnv               map[string][]string `json:"runtimeEnv,omitempty"`
+	Metadata                 map[string]string   `json:"metadata,omitempty"`
+	ShutdownAfterJobFinishes bool                `json:"shutdownAfterJobFinishes,omitempty"`
+	Entrypoint               string              `json:"entrypoint"`
+	JobId                    string              `json:"job_id,omitempty"`
+	RayClusterSpec           RayClusterSpec      `json:"rayClusterConfig,omitempty"`
+	// clusterSelector is used to select running rayclusters by labels
+	ClusterSelector map[string]string `json:"ClusterSelector,omitempty"`
+}
+
+// RayJobStatus defines the observed state of RayJob
+type RayJobStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	JobId          string    `json:"job_id,omitempty"`
+	RayClusterName string    `json:"rayClusterName,omitempty"`
+	JobStatus      JobStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// RayJob is the Schema for the rayjobs API
+type RayJob struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RayJobSpec   `json:"spec,omitempty"`
+	Status RayJobStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// RayJobList contains a list of RayJob
+type RayJobList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RayJob `json:"items"`
+}
+
+// RayJobInfo is the response of "ray job status" api.
+// Reference to https://docs.ray.io/en/master/cluster/jobs-package-ref.html#jobinfo.
+type RayJobInfo struct {
```

Comment:
Is it better to keep this file only for CRD related and move other struct into controller package?